.circleci/config.yml

```diff
@@ -72,55 +72,55 @@ commands:
           when: always
 
 jobs:
-  run-specs-with-postgres:
+  run-specs:
     executor:
-        name: solidusio_extensions/postgres
-        ruby_version: '2.7'
+        name: solidusio_extensions/<<parameters.database>>
+        ruby_version: <<parameters.ruby_version>>
     steps:
       - setup
       - test-branch:
-          branch: v3.1
-          rails_version: "~> 6.1"
-      - solidusio_extensions/store-test-results
-  run-specs-with-mysql:
-    executor:
-        name: solidusio_extensions/mysql
-        ruby_version: '3.0'
-    steps:
-      - setup
-      - test-branch:
-          branch: v3.1
-          rails_version: "~> 6.1"
-      - solidusio_extensions/store-test-results
-  run-specs-with-postgres-and-solidus-latest:
-    executor:
-        name: solidusio_extensions/postgres
-        ruby_version: '3.0'
-    steps:
-      - setup
-      - test-branch:
-          branch: master
-          rails_version: "~> 7.0"
-      - solidusio_extensions/store-test-results
-  run-specs-with-mysql-and-solidus-latest:
-    executor:
-        name: solidusio_extensions/mysql
-        ruby_version: '3.1'
-
-    steps:
-      - setup
-      - test-branch:
-          branch: master
-          rails_version: "~> 7.0"
+          branch: <<parameters.solidus_branch>>
+          rails_version: <<parameters.rails_version>>
       - solidusio_extensions/store-test-results
+    parameters:
+      solidus_branch:
+        type: string
+        default: 'v3.2'
+      rails_version:
+        type: string
+        default: '~> 7.0'
+      ruby_version:
+        type: string
+        default: '3.1'
+      database:
+        type: string
+        default: 'postgres'
 
 workflows:
   "Run specs on supported Solidus versions":
     jobs:
-      - run-specs-with-postgres
-      - run-specs-with-mysql
-      - run-specs-with-postgres-and-solidus-latest
-      - run-specs-with-mysql-and-solidus-latest
+      - run-specs
+
+      - run-specs:
+          name: run-specs-but-with-solidus-latest
+          solidus_branch: 'master'
+
+      - run-specs:
+          name: run-specs-but-with-rails-6
+          rails_version: '~> 6.1'
+
+      - run-specs:
+          name: run-specs-but-with-ruby-3-0
+          ruby_version: '3.0'
+
+      - run-specs:
+          name: run-specs-but-with-ruby-2
+          ruby_version: '2.7'
+
+      - run-specs:
+          name: run-specs-but-with-mysql
+          database: 'mysql'
```

Comment:
nit, non-mandatory: have you considered using a matrix setup with exclusions, it also supports dynamically composing the names based on parameters https://circleci.com/docs/configuration-reference#matrix-requires-version-21

Reply:
@elia Cool, it's the first time for me to see CircleCI matrices. However, having looked at the docs, I'm hesitant to switch to using matrices for this build:

* If we create a matrix for the 2 Solidus branches, 2 Rails versions, 3 Ruby versions, and 2 databases, we'll end up with 24 jobs, which seems excessive. Furthermore, we still have some flaky specs in our test suite right now, so a build with that many jobs would be unstable.
* If we use matrices but exclude everything to arrive at the 6-job config that we have now, we'll be excluding 18 combinations. 

I suppose we can review this if we find ourselves adding jobs to the build with more  complicated combinations of the parameters.

Your thoughts?